packages/client/mutations/useGenericMutation.ts

```diff
@@ -0,0 +1,48 @@
+import {useMutation, UseMutationConfig} from 'react-relay'
+import useAtmosphere from '../hooks/useAtmosphere'
+//import SendClientSegmentEventMutation from './SendClientSegmentEventMutation'
+
+type Handlers = {
+  onSuccess?: (res: any, variables: any) => void
+  onError?: (error: any) => void
+}
+
+export const useGenericMutation = (
+  mutation: any,
+  dataField: string,
+  successMessage: string,
+  errorMessage: string
+  // TODO some segment config?
+) => {
+  const [commit, submitting] = useMutation(mutation)
+  const atmosphere = useAtmosphere()
+
+  const execute = (config: UseMutationConfig<any>, handlers?: Handlers) => {
+    const {variables} = config
+    return commit({
+      onCompleted: (res: any) => {
+        const error = res[dataField]?.error
+
+        if (!error) {
+          atmosphere.eventEmitter.emit('addSnackbar', {
+            message: successMessage,
+            autoDismiss: 5,
+            key: `${dataField}Success`
+          })
+
+          handlers?.onSuccess && handlers.onSuccess(res, variables)
+        } else {
+          atmosphere.eventEmitter.emit('addSnackbar', {
+            message: errorMessage || error.message,
+            autoDismiss: 5,
+            key: `${dataField}Error`
+          })
+          handlers?.onError && handlers.onError(error)
+        }
+      },
```

Comment:
imo this assumes too much about the potential use cases. not every mutation wants to display toasts. if mutation wants to do that, it should provide a custom handler